estimate_infections_weekly_data.R

```diff
@@ -0,0 +1,140 @@
+# ESTIMATE INFECTIONS AND Rt WITH EPINOW2
+# ============================================================================== #
+
+# Extract parameters and maximum of the distribution for EpiNow2
+incubation_params <- epiparameter::get_parameters(covid_incubation_dist)
+incubation_max_days <- round(quantile(covid_incubation_dist, 0.999))  # Upper 99.9% range needed for EpiNow2
+
+# Create a LogNormal object for the incubation period
+incubation_lognormal <- EpiNow2::LogNormal(
+    meanlog = incubation_params[["meanlog"]],
+    sdlog = incubation_params[["sdlog"]],
+    max = incubation_max_days
+)
+
+# Extract parameters and maximum of the distribution for EpiNow2
+serial_interval_params <- epiparameter::get_parameters(serial_interval_dist)
+serial_interval_max_days <- round(quantile(serial_interval_dist, 0.999)) # Upper 99.9% range needed for EpiNow2
+
+# Create a LogNormal object for the serial interval
+serial_interval_lognormal <- EpiNow2::LogNormal(
+    meanlog = serial_interval_params[["meanlog"]],
+    sdlog = serial_interval_params[["sdlog"]],
+    max = serial_interval_max_days
+)
+
+# Create data with missing dates filled in for EpiNow2
+reported_cases_weekly_epinow <- fill_missing(
+    reported_cases_weekly,
+    missing_dates = "accumulate",
+    initial_accumulate = 1 # Don't model the first data point (to match EpiEstim method)
+)
+
+# Estimate infections using EpiNow2
+estimates_epinow <- epinow(
+    data = reported_cases_weekly_epinow,
+    generation_time = generation_time_opts(serial_interval_lognormal),
+    delays = delay_opts(incubation_lognormal),
+    forecast = forecast_opts(horizon = 0, accumulate = 1), # Forecasting is turned off to match with EpiEstim
+    rt = NULL, # Use the non-mechanistic model equivalent to EpiEstim
+    verbose = FALSE
+)
+
+# Initial look at the output
+plot(estimates_epinow$plots$R)
+
+# ============================================================================== #
+# ESTIMATE RT WITH EPIESTIM
+# ============================================================================== #
+
+# Prepare serial interval distribution. We'll reuse the serial interval distribution
+# extracted earlier.
+si_mean <- serial_interval_dist$summary_stats$mean
+si_sd <- serial_interval_dist$summary_stats$sd
+
+# Prepare the data by renaming the columns and converting the date to a Date object
+epiestim_data <- reported_cases_weekly |>
+    rename(I = confirm) |>
+    mutate(dates = as.Date(date), .keep = "unused") |>
+    as.data.frame()
+
+# Estimate infections and Rt using weekly aggregated data
+estimates_epiestim <- estimate_R(
+    incid = epiestim_data,
+    dt = 7L, # Aggregation window
+    dt_out = 7L, # Estimation rolling window
+    recon_opt = "naive",
+    method = "parametric_si",
+    config = make_config(list(mean_si = si_mean, std_si = si_sd))
+)
+
+# Initial look at the output
+plot(estimates_epiestim$R, "R") # Rt estimates only
```

Comment:
and after using incidence only, with this edit this line runs
```suggestion
plot(estimates_epiestim, "R") # Rt estimates only
```

Reply:
Thanks. See the updated script for the workaround with comment in https://github.com/epiverse-trace/epiverse_demos/pull/2#discussion_r1951725141.